### DISCUSSION TITLE
House load and direct solar consumption

### DISCUSSION THREAD

**Discussion by mjdyson** [johanmeijer/grott#119] [Q&A]
Hello,

Has anyone had any success in trying to create an instantaneous house consumption value from the data received from the inverter? I've been trying to create a visualisation of where my energy comes from (solar/grid/batteries).

My mix inverter provides a local load, but I am having trouble working out how this is calculated. Does this include grid import/export? Does it include self power consumption (i.e. the inverter energy use)?

Also I've been trying to create a separate visualisation for where my solar energy is going e.g. direct consumption, battery charging and exported. Again the numbers rarely seem to line up consistently. Has anyone had success in calculating an instantaneous direct solar consumption value?

Last question, what is meant by pvpowerout vs pvpowerin values? 

TIA :)

**Comment by johanmeijer** [comment]
Unluckily I do not have a mix inverter and can not help you here ;( Maybe others can. 

pvpowerin is the power direct from the solar panels (gross), pvpowerout is the power from the inverter to the grid (net). pvpowerout will be less because of the loss of (use of) energy in the inverter.

**Reply by mjdyson** [reply]
This is just a graphic for the instantaneous feeds. Nothing fancy here, but something doesn't add up how I expect. 
![IMG-20220515-WA0005.jpg](https://user-images.githubusercontent.com/4598068/168683283-52c7d5cc-7199-4bcc-9918-ed531fb89119.jpg)

**Reply by mjdyson** [reply]
And this is what I'm trying to achieve, but need the numbers to be more reliable first! 
![Screenshot_20220515-162048.png](https://user-images.githubusercontent.com/4598068/168683394-37c5eb1d-fb95-483f-af1b-7f2547434fe6.png)

**Reply by mjdyson** [reply]
And thanks for your reply. I suspect it might be a good idea to track the inverter consumption to see how much is being lost to the process. It would be interesting to see if this varies over time.
